packages/account/src/Sections/Verification/ProofOfAddress/__tests__/proof-of-address-form.spec.tsx

```diff
@@ -0,0 +1,101 @@
+import ProofOfAddressForm from '../proof-of-address-form';
+import { useDevice } from '@deriv-com/ui';
+import { mockStore, StoreProvider } from '@deriv/stores';
+import { BrowserRouter } from 'react-router-dom';
+
+jest.mock('@deriv-com/ui', () => ({
+    ...jest.requireActual('@deriv-com/ui'),
+    useDevice: jest.fn(),
+}));
+
+jest.mock('@deriv/hooks', () => ({
+    ...jest.requireActual('@deriv/hooks'),
+    useFileUploader: jest.fn(() => ({
+        upload: jest.fn(),
+    })),
+}));
+
+jest.mock('../poa-desktop-layout.tsx', () => jest.fn(() => 'mockedPOADesktopForm'));
+
+jest.mock('../poa-mobile-layout.tsx', () => jest.fn(() => 'mockedPOAMobileForm'));
+
+jest.mock('@deriv/shared', () => ({
+    ...jest.requireActual('@deriv/shared'),
+    WS: {
+        authorized: {
+            storage: {
+                getSettings: jest.fn().mockResolvedValue({
+                    get_settings: {
+                        address_line_1: 'test address_line_1',
+                        address_line_2: 'test address_line_2',
+                        address_city: 'test address_city',
+                        address_state: 'test address_state',
+                        address_postcode: 'test address_postcode',
+                    },
+                }),
+                getAccountStatus: jest.fn().mockResolvedValue({
+                    get_account_status: {
+                        authentication: {
+                            document: {
+                                status: 'none',
+                            },
+                            identity: {
+                                status: 'none',
+                            },
+                        },
+                    },
+                }),
+            },
+        },
+        setSettings: jest.fn(() => Promise.resolve({ error: '' })),
+        wait: jest.fn(() => Promise.resolve([])),
+        getSocket: jest.fn().mockReturnValue({}),
+    },
+}));
+
+describe('ProofOfAddressForm', () => {
+    const mock_props: React.ComponentProps<typeof ProofOfAddressForm> = {};
+
+    const mock_store = mockStore({
+        client: {
+            account_settings: {
+                address_line_1: 'test address_line_1',
+                address_line_2: 'test address_line_2',
+                address_city: 'test address_city',
+                address_state: 'test address_state',
+                address_postcode: 'test address_postcode',
+            },
+            fetchResidenceList: jest.fn(() => Promise.resolve('')),
+            fetchStatesList: jest.fn(() => Promise.resolve('')),
+            getChangeableFields: jest.fn(() => []),
+        },
+    });
+
+    const renderComponent = ({ props = mock_props, store = mock_store }) => {
+        return render(
+            <BrowserRouter>
+                <StoreProvider store={store}>
+                    <ProofOfAddressForm {...props} />
+                </StoreProvider>
+            </BrowserRouter>
+        );
+    };
```

Comment:
why don't we directly put `mock_props` and `mock_store` in the StoreProvider? So we can directly call renderComponent() without passing in empty object 🤔

Reply:
This was done so that if any testcase wants to customise the props, it can be done. Else the testcase will use the  default values. It could have been made as 2 props rather than a object.